### PR TITLE
ci: add reusable workflow for triggering packaging

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
In a subsequent PR, I'd like to use this in the release workflow triggered on tags. I've also explicitly set the downstream packaging workflow not to create a draft release.

### CODE REVIEW

File: .github/workflows/dev_builds.yml [hunk 1]
```diff
@@ -349,58 +349,12 @@ jobs:
...
-              "otc_sumo_version": "${{ steps.sumo-version.outputs.version }}"
-            }
-
-      # Outputs the Packaging Workflow URL as an Annotation in the UI after the
-      # packaging workflow has completed.
-      - name: Packaging Workflow URL
-        run: echo ::notice title=Packaging Workflow URL::${{ steps.trigger-packaging-workflow.outputs.workflow-url }}
+    uses: ./.github/workflows/workflow-trigger-packaging.yml
+    with:
+      create_release: false
```
Comment (rnishtala-sumo) [inline .github/workflows/dev_builds.yml:360]: So the dev builds were creating draft releases before correct?

File: .github/workflows/workflow-trigger-packaging.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+name: Trigger Packaging
+
+on:
+  workflow_call:
+    inputs:
+      create_release:
+        description: The packaging workflow will create a draft release in the packaging repository.
```
Comment (rnishtala-sumo) [inline .github/workflows/workflow-trigger-packaging.yml:7]: This only triggers packaging during a release build?

File: .github/workflows/workflow-trigger-packaging.yml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+          ref: main
+          wait-for-completion: false
+          display-workflow-run-url: true
+          display-workflow-run-url-interval: 1s
+          inputs: |-
+            {
+              "workflow_id": "${{ github.run_id }}",
+              "otc_version": "${{ steps.version-core.outputs.version }}",
+              "otc_sumo_version": "${{ steps.sumo-version.outputs.version }}"
+              "release": ${{ inputs.create_release }}
```
Comment (rnishtala-sumo) [inline .github/workflows/workflow-trigger-packaging.yml:59]: What does this additional "release" param used for?

File: .github/workflows/workflow-trigger-packaging.yml (hunk 2 above)
Comment (swiatekm-sumo) [reply .github/workflows/workflow-trigger-packaging.yml:7]: We don't trigger packaging at all for releases right now. I'd like to add that in a separate PR.

Comment (rnishtala-sumo) [reply .github/workflows/workflow-trigger-packaging.yml:7]: oh nvm, we seem to be using this during a dev build

File: .github/workflows/workflow-trigger-packaging.yml (hunk 3 above)
Comment (swiatekm-sumo) [reply .github/workflows/workflow-trigger-packaging.yml:59]: It controls whether we create a draft release in packaging: https://github.com/SumoLogic/sumologic-otel-collector-packaging/blob/1cf515c2b1bc92242cfcadaa61bf481cff188172/.github/workflows/build_packages.yml#L51

**Review by rnishtala-sumo — APPROVED** [review]

**Pull request merged**